### PR TITLE
Add module list command

### PR DESCRIPTION
**Pull request by pkratoch**
Closes: https://github.com/rpm-software-management/dnf5/issues/936
Closes: https://github.com/rpm-software-management/dnf5/issues/937

### CODE REVIEW

File: dnf5/commands/module/arguments.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public:
+    explicit ModuleDisabledOption(libdnf5::cli::session::Command & command)
+        : BoolOption(command, "disabled", '\0', _("Show disabled modules."), false) {}
+};
+
+
+class ModuleSpecArguments : public libdnf5::cli::session::StringArgumentList {
+public:
+    explicit ModuleSpecArguments(libdnf5::cli::session::Command & command)
+        : StringArgumentList(command, "module-spec", _("Pattern matching module NVSCAs.")) {}
```
Comment (jan-kolarik) [inline dnf5/commands/module/arguments.hpp:50]: Minor: typo NVSCA -> NSVCA

File: dnf5/commands/module/module_list.cpp [hunk 2]
```diff
@@ -19,13 +19,53 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
...
+    context.set_load_available_repos(Context::LoadAvailableRepos::ENABLED);
 }
 
-void ModuleListCommand::run() {}
+void ModuleListCommand::run() {
+    libdnf5::module::ModuleQuery query(get_context().base, true);
+    auto module_specs_str = module_specs->get_value();
+    if (module_specs_str.size() > 0) {
+        for (const auto & module_spec : module_specs_str) {
+            for (auto & nsvcap : libdnf5::module::Nsvcap::parse(module_spec)) {
```
Comment (jan-kolarik) [inline dnf5/commands/module/module_list.cpp:49]: I guess we could also use `const auto &` here ...

File: include/libdnf5-cli/output/modulelist.hpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            } else if (module_item.is_default()) {
+                stream_string.append(" [d]");
+            }
+
+            // Get profile strings (append [d] or [i] if needed)
+            std::vector<std::string> profile_strings;
+            for (auto profile : module_item.get_profiles()) {
+                // TODO(pkratoch): Also show "[i]" for installed profiles
+                // TODO(pkratoch): Make sure defaults are loaded
+                profile_strings.push_back(profile.is_default() ? profile.get_name() + " [d]" : profile.get_name());
```
Comment (jan-kolarik) [inline include/libdnf5-cli/output/modulelist.hpp:99]: Currently, the default profiles are not marked in the table, but I guess this is connected with the TODO comment above.

**Review by jan-kolarik — COMMENTED** [review]
Except for the mentioned comments above, it looks good to me.

File: dnf5/commands/module/arguments.hpp (hunk 1 above)
Comment (pkratoch) [reply dnf5/commands/module/arguments.hpp]: Fixed, thanks.

File: dnf5/commands/module/module_list.cpp (hunk 2 above)
Comment (pkratoch) [reply dnf5/commands/module/module_list.cpp]: Fixed, thanks.

File: include/libdnf5-cli/output/modulelist.hpp (hunk 3 above)
Comment (pkratoch) [reply include/libdnf5-cli/output/modulelist.hpp:99]: Yes, the TODO is there for this reason. I just found out that the defaults are loaded, they are just not accessed the correct way. I plan to fix that.

**Review by jan-kolarik — APPROVED** [review]

**Pull request merged**